### PR TITLE
Add Tiger910

### PR DESCRIPTION
**Pull request by Weirdo-F**
## Description
Add new keyboard:Tiger910

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tiger910/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#define _BASE 0
+#define _FN1 1
```
Comment (drashna) [inline keyboards/tiger910/keymaps/default/keymap.c:19]: We highly recommend enums over defines, as it's easier to maintain long term. 
```suggestion
enum custem_layers {
    _BASE,
    _FN1,
};
```

File: keyboards/tiger910/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+#define _BASE 0
+#define _FN1 1
+
```
Comment (drashna) [inline keyboards/tiger910/keymaps/via/keymap.c:21]: We highly recommend enums over defines, as it's easier to maintain long term. 
```suggestion
enum custem_layers {
    _BASE,
    _FN1,
};
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by Weirdo-F** [comment]
@drashna  what should I do next?

File: keyboards/tiger910/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ */
+#include QMK_KEYBOARD_H
+
+enum custem_layers {
+    _BASE,
+    _FN1,
+};
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```
Comment (drashna) [inline keyboards/tiger910/keymaps/via/keymap.c:24]: This needs to have four layers set.

File: keyboards/tiger910/readme.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+![Tiger910](https://www.hualigs.cn/image/60272570d2764.jpg)
+
+A 65% keyboard 
+
+* Keyboard Maintainer:  [Weirdo](https://weirdo-f.github.io)(https://github.com/1248314361)
+* Hardware Supported: Tiger910
+* Hardware Availability: Not yet
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/tiger910/readme.md:11]: Please add instructions on how to jump into the bootloader to flash new firmware.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/tiger910/keymaps/via/keymap.c (hunk 3 above)
Comment (Weirdo-F) [reply keyboards/tiger910/keymaps/via/keymap.c:26]: Modified

File: keyboards/tiger910/readme.md (hunk 4 above)
Comment (Weirdo-F) [reply keyboards/tiger910/readme.md:11]: Bootloader has been modified to Catalina. Do I need to add jump instructions?

File: keyboards/tiger910/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    _BASE,
+    _FN,
+    _FN1,
+    _FN2
+};
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT(
+    KC_ESC,   KC_1,    KC_2,    KC_3,   KC_4,   KC_5,    KC_6,    KC_7,   KC_8,    KC_9,     KC_0,     KC_MINS,   KC_EQL,    KC_BSPC,  KC_BSPC,  \
```
Comment (drashna) [inline keyboards/tiger910/keymaps/via/keymap.c:28]: The backslashes should be removed from the keymap. They're not actually needed (and are a hold over from older versions)

File: keyboards/tiger910/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    KC_LCTL,  KC_LGUI, KC_LALT,                 KC_SPC,  KC_SPC,          KC_SPC,            KC_RALT,  MO(_FN1),  KC_LEFT,   KC_DOWN,  KC_RIGHT  \
+  ),
+
+  [_FN1] = LAYOUT(
+    KC_GRV,   KC_F1,   KC_F2,   KC_F3,  KC_F4,  KC_F5,   KC_F6,   KC_F7,  KC_F8,   KC_F9,    KC_F10,   KC_F11,    KC_F12,    KC_DEL,    KC_DEL,  \
+    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS, \
+    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS, \
+    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS, \
+    KC_TRNS,  KC_TRNS, KC_TRNS,                 KC_TRNS, KC_TRNS,         KC_TRNS,           KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS  \
+  )
```
Comment (drashna) [inline keyboards/tiger910/keymaps/default/keymap.c:38]:
```suggestion
  [_BASE] = LAYOUT(
    KC_ESC,   KC_1,    KC_2,    KC_3,   KC_4,   KC_5,    KC_6,    KC_7,   KC_8,    KC_9,     KC_0,     KC_MINS,   KC_EQL,    KC_BSPC,  KC_BSPC,  
    KC_TAB,   KC_Q,    KC_W,    KC_E,   KC_R,   KC_T,    KC_Y,    KC_U,   KC_I,    KC_O,     KC_P,     KC_LBRC,   KC_RBRC,   KC_BSLS,  KC_DEL,   
    KC_CAPS,  KC_A,    KC_S,    KC_D,   KC_F,   KC_G,    KC_H,    KC_J,   KC_K,    KC_L,     KC_SCLN,  KC_QUOT,   KC_BSLS,   KC_ENT,   KC_PGUP,  
    KC_LSFT,  KC_LSFT, KC_Z,    KC_X,   KC_C,   KC_V,    KC_B,    KC_N,   KC_M,    KC_COMM,  KC_DOT,   KC_SLSH,   KC_RSFT,   KC_UP,    KC_PGDN,  
    KC_LCTL,  KC_LGUI, KC_LALT,                 KC_SPC,  KC_SPC,          KC_SPC,            KC_RALT,  MO(_FN1),  KC_LEFT,   KC_DOWN,  KC_RIGHT  
  ),

  [_FN1] = LAYOUT(
    KC_GRV,   KC_F1,   KC_F2,   KC_F3,  KC_F4,  KC_F5,   KC_F6,   KC_F7,  KC_F8,   KC_F9,    KC_F10,   KC_F11,    KC_F12,    KC_DEL,    KC_DEL,  
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS, 
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS, 
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS, 
    KC_TRNS,  KC_TRNS, KC_TRNS,                 KC_TRNS, KC_TRNS,         KC_TRNS,           KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS  
  )
```

File: keyboards/tiger910/keymaps/via/keymap.c (hunk 5 above)
Comment (Weirdo-F) [reply keyboards/tiger910/keymaps/via/keymap.c]: OK

Comment (Weirdo-F) [reply keyboards/tiger910/keymaps/via/keymap.c]: OK

File: keyboards/tiger910/readme.md (hunk 4 above)
Comment (drashna) [reply keyboards/tiger910/readme.md:11]: Yes.  Specifically, the physical "how".  As in, is there a hardware button?  is bootmagic enabled?  location for RESET keycode.  Etc.

Comment (Weirdo-F) [reply keyboards/tiger910/readme.md:11]: ok，I have made changes.

File: keyboards/tiger910/info.json [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
```
Comment (drashna) [inline keyboards/tiger910/info.json:15]: I don't think that comments are supported in json like this. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/tiger910/info.json (hunk 7 above)
Comment (Weirdo-F) [reply keyboards/tiger910/info.json]: ok.

**Review by drashna — APPROVED** [review]

File: keyboards/tiger910/tiger910.h [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT( \
```
Comment (fauxpark) [inline keyboards/tiger910/tiger910.h:21]: This file doesn't match the info.json.

File: keyboards/tiger910/rules.mk [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = no	      # USB Nkey Rollover
+RGBLIGHT_ENABLE = no
+
```
Comment (fauxpark) [inline keyboards/tiger910/rules.mk:18]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/tiger910/tiger910.h (hunk 8 above)
Comment (Weirdo-F) [reply keyboards/tiger910/tiger910.h:21]: OK, I have made corresponding changes.

Comment (Weirdo-F) [reply keyboards/tiger910/tiger910.h:21]: Please check again if there is any need to modify

File: keyboards/tiger910/config.h [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+
+
+
+
```
Comment (fauxpark) [inline keyboards/tiger910/config.h:54]:
```suggestion
```

File: keyboards/tiger910/keymaps/via/keymap.c (hunk 3 above)
Comment (fauxpark) [reply keyboards/tiger910/keymaps/via/keymap.c:18]: Still outstanding.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/tiger910/keymaps/via/keymap.c (hunk 3 above)
Comment (Weirdo-F) [reply keyboards/tiger910/keymaps/via/keymap.c]: Modified

**Comment by Weirdo-F** [comment]
@fauxpark what should I do next?

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**